crates/sui-framework/sources/governance/validator.move

```diff
@@ -74,10 +78,30 @@ module sui::validator {
         gas_price: u64,
     }
 
+    const PROOF_OF_POSSESSION_DOMAIN: vector<u8> = vector[107, 111, 115, 107];
+
+    fun verify_proof_of_possession(
+        proof_of_possession: vector<u8>,
+        sui_address: address,
+        pubkey_bytes: vector<u8>
+    ) {
+        // The proof of possession is the signature over ValidatorPK || AccountAddress.
+        // This proves that the account address is owned by the holder of ValidatorPK, and ensures
+        // that PK exists.
+        let signed_bytes = pubkey_bytes;
+        let address_bytes = bcs::to_bytes(&sui_address);
+        vector::append(&mut signed_bytes, address_bytes);
```

Comment:
The only reason this is secure is because address has a fixed size. Not required in this PR, but when intent signing crosses the line, we should consider applying it here as well.